spec/unit/rust-crypto/CrossSigningInfo.spec.ts

```diff
@@ -0,0 +1,114 @@
+/*
+Copyright 2023 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import fetchMock from "fetch-mock-jest";
+
+import { CrossSigningInfoImpl } from "../../../src/rust-crypto/CrossSigningInfoImpl";
+import { SIGNED_CROSS_SIGNING_KEYS_DATA, TEST_USER_ID } from "../../test-utils/test-data";
+import { CrossSigningKey } from "../../../src/crypto-api";
+import { HttpApiEvent, HttpApiEventHandlerMap, IHttpOpts, MatrixHttpApi, TypedEventEmitter } from "../../../src";
+
+describe("CrossSigningInfo", () => {
+    let httpApi: MatrixHttpApi<IHttpOpts & { onlyData: true }>;
+
+    beforeEach(async () => {
+        const dummyEventEmitter = new TypedEventEmitter<HttpApiEvent, HttpApiEventHandlerMap>();
+        httpApi = new MatrixHttpApi(dummyEventEmitter, {
+            baseUrl: "https://example.com",
+            prefix: "/_matrix",
+            onlyData: true,
+        });
+    });
+
+    afterEach(() => fetchMock.mockReset());
+
+    describe("CrossSigningInfo.create", () => {
+        it("should return null when the master keys are missing", async () => {
+            fetchMock.post("https://example.com/_matrix/client/v3/keys/query", {});
+
+            const crossSigningInfo = await CrossSigningInfoImpl.create(TEST_USER_ID, httpApi);
+            expect(crossSigningInfo).toBeNull();
+        });
+
+        it("should return null when the self signing keys are missing", async () => {
+            fetchMock.post("https://example.com/_matrix/client/v3/keys/query", {
+                master_keys: {
+                    [TEST_USER_ID]: {
+                        keys: {
+                            "ed25519:pubKey": "pubKey",
+                        },
+                    },
+                },
+            });
+
+            const crossSigningInfo = await CrossSigningInfoImpl.create(TEST_USER_ID, httpApi);
+            expect(crossSigningInfo).toBeNull();
+        });
+
+        it("should return null when the user signing keys are missing", async () => {
```

Comment:
It is *normal* for the `user_signing_keys` property to be *present*, but empty. The user-signing key is for the user's own use, and is not normally shared with other users.

So I think (a)  this test is testing the wrong thing; (b) it is expecting the wrong result.